plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsViewModel.java

```diff
@@ -92,7 +95,11 @@ protected void update(Connection connection, List<Connection> connections,
 		}
 		deployProject = updateDeployProject(deployProject, projects);
 		updateSourcePath(sourcePath, deployProject);
-		updatePodPath(podPath, deploymentMapperByProjectName, service);
+		if(serviceOrDeploymentChanged) {
```

Comment:
This is version before Andre's commit JBIDE-21452:
protected void updatePodPath(String newPodPath) {
		if (newPodPath == null) {
			newPodPath = "/opt/app-root/src";
		}
		firePropertyChange(PROPERTY_POD_PATH, this.podPath, this.podPath = newPodPath);
	}
This is after:

	protected void updatePodPath(String newPodPath, Map<String, DeploymentResourceMapper> deploymentResourceMapperByProjectName, IService service) {
		this.deploymentMapperByProjectName = deploymentResourceMapperByProjectName;
		newPodPath = getDeploymentDirectory(service, deploymentResourceMapperByProjectName);
		firePropertyChange(PROPERTY_POD_PATH, this.podPath, this.podPath = newPodPath);
	}
Look like Andre forgot check newPodPath == null. Can you please ask him?